GowerMetric.py

```diff
@@ -154,16 +230,88 @@ def fit(self, X):
                     ratio_cols[:, zero_values_mask], axis=0
                 )
 
-            if self.ratio_scale_normalization == "kde":
-                n = X.shape[0]
-                c = 1.06
-
-                s = np.std(ratio_cols, axis=0)
-                self.h_ = (
-                    c / n ** (1 / 5) * np.min([s, self.ranges_ / 1.34], axis=0)
-                )
-
-        loader = GowerMetricWeights(self)
+            n = X.shape[0]
+
+            # h_t parameter
+            if self.ratio_scale_window == "kde":
+                if self.kde_type == "silverman":
+                    c = 1.06
+
+                    s = np.std(ratio_cols, axis=0)
+                    self.h_ = (
+                        c
+                        / n ** (1 / 5)
+                        * np.min([s, self.ranges_ / 1.34], axis=0)
+                    )
+                elif self.kde_type == "scott":
+                    c = 0.9
+
+                    s = np.std(ratio_cols, axis=0)
+                    self.h_ = c / n ** (1 / 5) * s
+                elif self.kde_type == "sheather-jones":
+                    from KDEpy import FFTKDE
+
+                    try:
+                        self.h_ = np.array(
+                            [
+                                FFTKDE(kernel="gaussian", bw="ISJ")
```

Comment:
I think you can create this object before the loop and then just use it to fit and get `.bw` in list comprehension.